core/lib/spree/core/controller_helpers/pricing.rb

```diff
@@ -0,0 +1,17 @@
+module Spree
+  module Core
+    module ControllerHelpers
+      module Pricing
+        extend ActiveSupport::Concern
+
+        included do
+          helper_method :current_currency
+        end
+
+        def current_currency
+          current_store.try(:default_currency).presence || Spree::Config[:currency]
```

Comment:
I took some specs from `solidus_multi_domain`, and this case has been explicitly tested for: https://github.com/solidusio/solidus/pull/1060/files#diff-14ee7de4c10a42a47306354bec6b902eR20